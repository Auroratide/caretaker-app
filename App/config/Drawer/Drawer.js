import React from 'react';
import { DrawerItems } from 'react-navigation';
import { ThemeProvider, Drawer as MuiDrawer } from 'react-native-material-ui';
import theme from '../theme';

export default function Drawer( { navigation, items, screenProps } ) {
  const drawerItems = items.map((item) => {
    return {
      icon: screenProps[item.key].navigationOptions.drawerIcon,
      value: screenProps[item.key].navigationOptions.drawerLabel,
      onPress: () => navigation.navigate(item.routeName)
    };
  });

  drawerItems[navigation.state.index].active = true;

  return <ThemeProvider uiTheme={theme}>
    <MuiDrawer>
      <MuiDrawer.Section divider title='Caretaker' />
      <MuiDrawer.Section items={drawerItems} />
    </MuiDrawer>
  </ThemeProvider>;
}
