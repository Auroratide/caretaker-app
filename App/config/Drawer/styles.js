import { StyleSheet } from 'react-native';

export default styles = {
  header: {
    contentContainer: {
      backgroundColor: '#EEEEEE'
    }
  }
};
