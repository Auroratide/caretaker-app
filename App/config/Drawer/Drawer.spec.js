import React from 'react';
import renderer from 'react-test-renderer';
import Drawer from './Drawer';

describe('<Drawer />', () => {
  let props;
  const drawer = () => renderer.create(<Drawer {...props} />);

  beforeEach(() => {
    props = {
      navigation: undefined,
      items: undefined,
      screenProps: undefined
    };
  });

  it('should render correctly', () => {
    props.navigation = {
      navigate: jest.fn(),
      state: {
        index: 0
      }
    };

    props.items = [ {
      key: 'Hello'
    }, {
      key: 'Goodbye'
    } ];

    props.screenProps = {
      Hello: {
        navigationOptions: {
          drawerIcon: 'hey-there',
          drawerLabel: 'Greetings and Salutations'
        }
      },
      Goodbye: {
        navigationOptions: {
          drawerIcon: 'so-long',
          drawerLabel: 'See Ya Later'
        }
      }
    };

    expect(drawer()).toMatchSnapshot();
  });

});
