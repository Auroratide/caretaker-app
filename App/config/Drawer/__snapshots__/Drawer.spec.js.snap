// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer /> should render correctly 1`] = `
<ThemeProvider
  uiTheme={
    Object {
      "palette": Object {
        "primaryColor": "#FF7575",
        "secondaryColor": "#E66969",
      },
      "toolbar": Object {
        "container": Object {
          "height": 50,
        },
      },
    }
  }
/>
`;
